spec/helpers/admin/elements_helper_spec.rb

```diff
@@ -18,46 +18,124 @@ module Alchemy
+          let(:page_definition) do
+            {'name' => "foo", "elements" => ["1", "2"]}
+          end
+
+          it "returns an empty string" do
+            expect(helper.grouped_elements_for_select(elements)).to eq("")
+          end
+        end
+
+        context "with empty cells in page definition" do
+          let(:page_definition) do
+            { "name" => "foo", "cells" => [], "elements" => ["1", "2"]}
+          end
+
+          it "returns an empty string" do
+            expect(helper.grouped_elements_for_select(elements)).to eq("")
+          end
+        end
+
+        context "with a cell containing no elements" do
+          let(:cell_definitions) do
+            [{"name" => "empty_cell", "elements" => []}]
+          end
+
+          let(:page_definition) do
+            {"name" => "foo", "cells" => ["empty_cell"], "elements" => ["1", "2"]}
+          end
+
+          it "does not include that cell" do
+            expect(helper.grouped_elements_for_select(elements)).to eq("Main content" => [["1", "1"], ["2", "2"]])
+          end
+        end
+
+        context "with an element in a cell only" do
+          let(:elements) do
+            [mock_model('Element', name: 'in_cell', display_name: 'In Cell')]
+          end
+
+          let(:page_definition) do
+            { 'name' => "foo", 'cells' => ["foo_cell"], 'elements' => [] }
+          end
+
+          let(:cell_definitions) do
+            [{'name' => "foo_cell", 'elements' => ["in_cell"]}]
+          end
+
+          it "returns an option for the element in the cell only" do
+            expect(helper.grouped_elements_for_select(elements)).to include("Foo cell" => [["In cell", "in_cell#foo_cell"]])
+          end
         end
-      end
 
-      context "with cell having no elements" do
-        it "should remove that cell from hash" do
-          expect(helper.grouped_elements_for_select(elements)['Empty cell']).to be_nil
+        context "with the same element in both cell and page" do
+          let(:element_definitions) do
+            [{"name" => "in_cell_and_page"}]
+          end
+
+          let(:elements) do
+            [mock_model('Element', name: 'in_cell_and_page', display_name: 'In Cell and Page')]
+          end
+
+          let(:page_definition) do
+            {
+              'name' => "foo",
+              'cells' => ["foo_cell"],
+              'elements' => ["in_cell_and_page"]
+            }
+          end
+
+          let(:cell_definitions) do
+            [{
+              'name' => "foo_cell",
+              'elements' => ["in_cell_and_page"]
+            }]
+          end
+
+          it "returns two options of same element, one for the cell and one for the page" do
+            expect(helper.grouped_elements_for_select(elements)).to eq({
+              "Main content" => [["In cell and page", "in_cell_and_page"]],
+              "Foo cell" => [["In cell and page", "in_cell_and_page#foo_cell"]]
+            })
```

Comment:
Indent the right brace the same as the first position after the preceding left parenthesis.